### PR TITLE
Extend erl_call with a new flag to access a node at HOSTNAME:PORT

### PR DESCRIPTION
**Pull request by kjellwinblad**
This pull request adds the flag `-address` to the `erl_call` program. This flag makes it possible for the user to use `erl_call` to interact with a node even if no EPMD instance is running on the node's host.

This pull request also adds a family of new functions to the `erl_interface` API (that are needed implement the new `erl_call` flag and similar types of functionality) and extends the documentation of `erl_call` and `erl_interface` with descriptions of the new flag and functions.

### CODE REVIEW

**Comment by garazdawi** [comment]
Please split this into two commits, one for ei and one for erl_call.

File: lib/erl_interface/doc/src/ei_connect.xml [hunk 1]
```diff
@@ -422,20 +422,30 @@ typedef struct {
     <func>
       <name since=""><ret>int</ret><nametext>ei_connect(ei_cnode* ec, char *nodename)</nametext></name>
       <name since=""><ret>int</ret><nametext>ei_xconnect(ei_cnode* ec, Erl_IpAddr adr, char *alivename)</nametext></name>
+      <name since="OTP 23"><ret>int</ret><nametext>ei_connect_host_port(ei_cnode* ec, char *hostname, int port)</nametext></name>
+      <name since="OTP 23"><ret>int</ret><nametext>ei_xconnect_host_port(ei_cnode* ec, Erl_IpAddr adr, int port)</nametext></name>
```
Comment (sverker) [inline lib/erl_interface/doc/src/ei_connect.xml]: Use ticket syntax: since="OTP @OTP-12345@"

File: lib/erl_interface/doc/src/ei_connect.xml [hunk 2]
```diff
@@ -571,13 +581,16 @@ if (ei_connect_init(&ec, "madonna", "cookie...", n++) < 0) {
     <func>
       <name since=""><ret>int</ret><nametext>ei_connect_tmo(ei_cnode* ec, char *nodename, unsigned timeout_ms)</nametext></name>
       <name since=""><ret>int</ret><nametext>ei_xconnect_tmo(ei_cnode* ec, Erl_IpAddr adr, char *alivename, unsigned timeout_ms)</nametext></name>
+      <name since="OTP 23"><ret>int</ret><nametext>ei_connect_host_port_tmo(ei_cnode* ec, char *hostname, int port, unsigned ms)</nametext></name>
+      <name since="OTP 23"><ret>int</ret><nametext>ei_xconnect_host_port_tmo(ei_cnode* ec, Erl_IpAddr adr, int port, unsigned ms)</nametext></name>
```
Comment (sverker) [inline lib/erl_interface/doc/src/ei_connect.xml]: and here

File: lib/erl_interface/doc/src/ei_connect.xml (hunk 1 above)
Comment (kjellwinblad) [reply lib/erl_interface/doc/src/ei_connect.xml]: Thanks, done in commit below.

File: lib/erl_interface/doc/src/ei_connect.xml (hunk 2 above)
Comment (kjellwinblad) [reply lib/erl_interface/doc/src/ei_connect.xml]: Thanks, done in commit below.

**Comment by kjellwinblad** [comment]
> Please split this into two commits, one for ei and one for erl_call.

I have done so and fixed a memory issue and a compile problem on windows.

**Pull request merged**